sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -46,20 +46,62 @@ const debounce = (func, wait) => {
 /**
  * Take an object as parameter and convert it to
  * url params string.
- * Eg. if obj = { 'a': 1, 'b': 2 }, then it will return
- * the string a=1&b=2.
+ * Eg. if obj = { 'a': 1, 'b': 2, 'c': ['hello', 'world'] }, then it will return
+ * the string a=1&b=2&c=hello,world
  *
  * @param {Object} obj the object to be converted
- * @return {String} object in url params form
+ * @return {String|Array} object in url params form
  */
 const convertObjToUrlParams = obj => {
-    const params = Object.keys(obj)
-        .map(function(key) {
-            const s = key + "=" + obj[key];
+    let params = Object.keys(obj).map(function(key) {
+        if (_is_string(key)) {
+            const s = key + "=" + encodeURI(obj[key]);
             return s;
-        })
-        .join("&");
-    return params;
+        }
+    });
+
+    // removing empty strings from the 'params' array
+    let final_params = [];
+    for (let i = 0; i < params.length; ++i) {
+        if (_is_string(params[i])) {
+            final_params.push(params[i]);
+        }
+    }
+    if (final_params.length === 1) {
+        return final_params[0];
+    } else {
+        let final_url_params = final_params.join("&");
+        return final_url_params;
+    }
+};
+
+/**
+ * Adds/removes "rtd_search" url parameter to the url.
+ */
+const updateUrl = () => {
+    let origin = window.location.origin;
+    let path = window.location.pathname;
+    let url_params = $.getQueryParameters();
+
+    if (_is_string(SEARCH_QUERY)) {
```

Comment:
This will always be a string. It will become false if `SEARCH_QUERY` becomes empty string.
`_is_string` func checks two things:
- The type of the data
- if the type is 'string', then it should be of length >= 1